R/get_meetnetten_locations.R

```diff
@@ -0,0 +1,178 @@
+#'    }
+#' \item \code{sublocations}: sf object of the the sublocations (for example
+#' the sections of a transect) for each of the selected main locations, with
+#' following attribute variables:
+#'    \itemize{
+#'    \item \code{species_group}
+#'    \item \code{scheme}: name of the monitoring scheme
+#'    \item \code{location}: name of the main location
+#'    \item \code{sublocation}: name of the sublocation
+#'    \item \code{is_active}: whether the sublocation is counted or not
+#'    }
+#' }
+#'
+#' Not all main locations are subdivided in sublocations.
+#' So in some cases the sublocations object is empty.
+#'
+#' @details
+#' Each monitoring scheme of the species monitoring programme of Flanders
+#' \href{www.meetnetten.be}{Meetnetten} consists of a fixed set of locations.
+#' A monitoring scheme for rare species includes all locations where the species
+#' occurs. For more common species a sample of locations is drawn and the
+#' the selected locations are included in the monitoring scheme. In some cases,
+#' the monitoring project in \href{www.meetnetten.be}{Meetnetten} also contains
+#' locations that are not part of the sample. These locations can be counted
+#' optionally and are indicated by (\code{is_sample} = \code{FALSE}).
+#'
+#' It also occurs that a location becomes inaccessible or that the target
+#' species disappears. Then, a locations can be made inactive (\code{is_active}
+#' = \code{FALSE}), which means that no observations can be recorded any more.
+#'
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr filter arrange collect tbl sql mutate select
+#' @importFrom stringr str_to_lower
+#' @importFrom sf st_as_sf st_drop_geometry
+#'
+#' @export
+#' @family meetnetten
+#' @examples
+#' \dontrun{
+#' library(inbodb)
+#' con <- connect_inbo_dbase("S0008_00_Meetnetten")
+#'
+#' # get locations for a specific monitoring scheme
+#' locations_heivlinder <- get_meetnetten_locations(con,
+#'                                                  scheme_name = "Heivlinder")
+#'
+#' main_locations <- locations_heivlinder$main_locations
+#' sublocations <- locations_heivlinder$sublocations
+#'
+#' # get locations for a specific species_group
+#' locations_dragonflies <- get_meetnetten_locations(con,
+#'                                                   species_group = "libellen")
+#'
+#' main_locations <- locations_dragonflies$main_locations
+#' sublocations <- locations_dragonflies$sublocations
+#'
+#' # Close the connection when done
+#' dbDisconnect(con)
+#' rm(con)
+#' }
+
+get_meetnetten_locations <- function(connection,
+                                    scheme_name = NULL,
+                                    species_group = NULL) {
+
+  species_group_selected <- species_group
+
+  assert_that(inherits(connection, what = "Microsoft SQL Server"),
+              msg = "Not a connection object to database.")
+
+  if (!is.null(scheme_name)) {
+    assert_that(is.character(scheme_name))
```

Comment:
Unless you want to select multiple schemas use this. 
```suggestion
    assert_that(is.string(scheme_name), noNA(schema_name))
```

Otherwise use
```suggestion
    assert_that(is.character(scheme_name), length(scheme_name) > 0, noNA(scheme_name))
```